### PR TITLE
Product Creation AI: UI for preview screen

### LINKED ISSUE #10750: Product Creation AI: UI for preview screen
image

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #10750

## Description

This PR adds UI for the preview screen of the product creation AI flow and updates navigation to display the screen.

## Testing instructions

- Log in to a WPCom store or a self-hosted store with Jetpack AI installed.
- Switch to Products tab and select "+" to add a new product.
- Select add product with AI.
- Enter product name and features to get to the preview screen.
- Notice that the UI matches the design. The values will need to be updated in a subsequent PR.

## Screenshots

| \ | Top details | Bottom details | Loading |
| - | - | - | - |
| Light mode |  |  |  |
| Dark mode |  |  |  |

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    enum Constants {
+        static let dummyText = "Lorem ipsum dolor sit amet, consectetur adipiscing elit"
+    }
+    enum Localization {
+        static let title = NSLocalizedString(
+            "Preview",
+            comment: "Title on the add product with AI Preview screen."
+        )
+        static let subtitle = NSLocalizedString(
+            "Don't worry. You can always change those details later.",
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift]: > "Don't worry. You can always change those details later."

Opinion: I think changing "those" into "below"/"these" will add more clarity.

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+                    .padding(.bottom, Layout.contentPadding)
+
+                    VStack(spacing: 0) {
+                        TitleAndValueDetailRow(title: Localization.price,
+                                               value: "Regular price: $15",
+                                               image: UIImage.priceImage,
+                                               isLoading: viewModel.isGeneratingDetails)
+                        Divider()
+                            .background(Color(.separator))
+                        TitleAndValueDetailRow(title: Localization.inventory,
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift:72]: Opinion: Should inventory be part of the preview screen? I don't think we will generate anything related to stock status using AI.

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+                                               value: Localization.inStock,
+                                               image: UIImage.inventoryImage,
+                                               isLoading: viewModel.isGeneratingDetails)
+                        Divider()
+                            .background(Color(.separator))
+                        TitleAndValueDetailRow(title: Localization.categories,
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift:80]: Super nit: Consider adding comments/empty lines between each field for readability.

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+                })
+                .renderedIf(viewModel.isGeneratingDetails == false)
+            }
+            .padding(insets: Layout.insets)
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    if viewModel.isGeneratingDetails {
+                        ActivityIndicator(isAnimating: .constant(true), style: .medium)
+                    } else {
+                        Button("Save as draft") {
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift]: Super nit: I think we can localize this string.

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+        let dummyText = ProductDetailPreviewView.Constants.dummyText
+
+        typealias Layout = ProductDetailPreviewView.Layout
+
+        var body: some View {
+            Text(content ?? dummyText)
+                .bodyStyle()
+                .multilineTextAlignment(.leading)
+                .padding(Layout.contentPadding)
+                .frame(maxWidth: .infinity, alignment: .leading)
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift:148]: ❓ For my learning: What is the purpose of using a `.frame` modifier here? Thank you!

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+                .background(Color(.init(light: .systemGray6, dark: .tertiarySystemBackground)))
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift]: ❓ What do you think about declaring this color as a constant and reusing it in `BasicDetailRow` and `TitleAndValueDetailRow`?

**Review by selanthiraiyan — APPROVED** [review]
This screen looks great. 🚀 

I left a few non-blocking questions.

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift]: I was unsure about this wording too. Asking Ann in the design, will update this later 👍

Comment (annchichi) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift]: I've changed it to "below" as it surely gains more clarity. :) Thanks!

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift (hunk 2 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift:72]: I see this on the design, my understanding is that we can add that extra information for the product ourselves as we expect the new product to be in stock. Does that make sense? I'll ping the team about that too.

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift (hunk 5 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift:148]: If the text is not long enough, I noticed that the frame's width does not fit the whole screen. I tried `.fixedSize` but it didn't work, so I had to use this.

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift]: Updated in 238a341447d96be21005e962f45493e915053404.

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift (hunk 3 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift:80]: Updated in 399404cc3eb5aeb0008811fb34eccf2a9f507501.

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift (hunk 4 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift]: Updated in a70e2e2ccc0f45410b9c1df3e0e67ab864968e94.

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift (hunk 6 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift]: Done in c779d56363fd6076aec4a0d6342de1ecca54a2a7.

**Comment by itsmeichigo** [comment]
Thanks Sharma for the reviews 🙇‍♀️ I'm merging this PR now to unblock development, please feel free to leave questions if needed.

**Pull request merged**